ngraph/core/src/op/dft.cpp

```diff
@@ -0,0 +1,259 @@
+    {
+        NODE_VALIDATION_CHECK(this,
+                              axes_shape.rank().get_length() == 1,
+                              "DFT axes input must be 1D tensor. Got axes input rank: ",
+                              axes_shape.rank().get_length());
+    }
+
+    if (input_shape.rank().is_static() && axes_shape.is_static())
+    {
+        const auto input_rank = input_shape.rank().get_length();
+        NODE_VALIDATION_CHECK(this,
+                              input_rank >= axes_shape.to_shape()[0] + 1,
+                              "The input rank must be greater than number of DFT axes. Got "
+                              "input rank: ",
+                              input_rank,
+                              ", number of axes: ",
+                              axes_shape.to_shape()[0]);
+    }
+
+    if (input_shape.rank().is_static() && is_type<op::Constant>(input_value(1).get_node()))
+    {
+        const auto input_rank = input_shape.rank().get_length();
+        const auto& const_axes = get_constant_from_source(input_value(1));
+        auto axes = const_axes->cast_vector<int64_t>();
+
+        for (int64_t& axis : axes)
+        {
+            if (axis < 0)
+            {
+                axis += input_rank - 1;
+            }
+        }
+
+        AxisVector axes_vector;
+        AxisSet axes_set;
+        for (const int64_t axis : axes)
+        {
+            axes_vector.push_back(static_cast<size_t>(axis));
+            axes_set.insert(static_cast<size_t>(axis));
+        }
+
+        NODE_VALIDATION_CHECK(
+            this, axes.size() == axes_set.size(), "DFT axes must be unique. Got: ", axes_vector);
+
+        NODE_VALIDATION_CHECK(this,
+                              std::find(axes.begin(), axes.end(), input_rank - 1) == axes.end(),
+                              "DFT axes cannot contain the last axis. Got axes: ",
+                              axes_vector);
+    }
+
+    if (num_of_inputs == 3)
+    {
+        element::Type signal_size_et = get_input_element_type(2);
+        NODE_VALIDATION_CHECK(this,
+                              signal_size_et == element::i64 || signal_size_et == element::i32,
+                              "DFT signal_size element type must be i32 or i64");
+
+        PartialShape signal_size_shape = PartialShape(get_input_partial_shape(2));
+        if (signal_size_shape.rank().is_static())
+        {
+            NODE_VALIDATION_CHECK(this,
+                                  signal_size_shape.rank().get_length() == 1,
+                                  "DFT Signal size input must be 1D tensor. Got signal size "
+                                  "input rank: ",
+                                  signal_size_shape.rank().get_length());
+        }
+
+        if (axes_shape.is_static() && signal_size_shape.is_static())
+        {
+            NODE_VALIDATION_CHECK(this,
+                                  axes_shape.to_shape()[0] == signal_size_shape.to_shape()[0],
```

Comment:
Hmmm. Why do you compare the first elements? May be you wanted to compare `axes_shape.rank()` and `signal_size_shape .rank()`?

Reply:
The comparison `axes_shape.to_shape()[0] == signal_size_shape.to_shape()[0]` is correct, because:

1. at lines 105--111 we checked that `axes_shape.rank() == 1`;
2. at lines 164--171 we checked that `signal_size_shape.rank() == 1`.

Hence, at line 175, we have `axes_shape.rank() == 1 && signal_size_shape.rank() == 1`, that is ranks are equal.

Next, according to the DFT operation specification, 
1. `axes` - 1D tensor of type *T_IND* specifying dimension indices where DFT is applied;
2. `signal_size` - 1D tensor of type *T_SIZE* describing signal size with respect to axes from the input `axes`;
3. If the input `signal_size` is specified, the size of `signal_size` must be the same as the size of `axes`.